File: examples/java-junit-gradle/src/test/java/examples/YahtzeeSteps.java
package examples;

import dev.varar.Registrar;
import dev.varar.State;
import dev.varar.StateBinder;
import dev.varar.StepDefinitions;
import java.util.Arrays;
import java.util.List;
import java.util.Map;

public final class YahtzeeSteps implements StepDefinitions {

    record Ctx() implements State {}

    @Override
    public void defineSteps(Registrar registrar) {
        StateBinder<Ctx> s = registrar.steps(Ctx::new);

        // Header-bound table: the paragraph names every header cell (dice,
        // category, score), so this sensor runs once per row with the row as a
        // Map keyed by header. Returning Map.of("score", …) checks that column;
        // the other columns are inputs.
        s.sensor("Examples of dice, category and score", (Ctx ctx, Map<String, String> row) -> {
            List<Integer> dice = Arrays.stream(row.get("dice").split(","))
                    .map(d -> Integer.parseInt(d.trim()))
                    .toList();
            return Map.of("score", Yahtzee.score(dice, row.get("category")));
        });
    }
}

File: examples/kotlin-junit/src/test/kotlin/examples/yahtzee.steps.kt
@file:JvmName("YahtzeeSteps")

package examples

import dev.varar.kotlin.sensor
import dev.varar.kotlin.steps

val yahtzeeSteps = steps {
    // Header-bound table: the paragraph names every header cell (dice,
    // category, score), so this sensor runs once per row with the row as a
    // Map keyed by header. Returning mapOf("score" to …) checks that column;
    // the other columns are inputs.
    sensor("Examples of dice, category and score") { row: Map<String, String> ->
        val dice = row.getValue("dice").split(",").map { it.trim().toInt() }
        mapOf("score" to score(dice, row.getValue("category")))
    }
}
